src/pipeline.cc

```diff
@@ -361,8 +366,31 @@ bool Indexer_Parse(SemaManager *completion, WorkingFiles *wfiles,
   return true;
 }
 
+void Quit(SemaManager &manager) {
+  quit = true;
+  manager.Quit();
+
+  { std::lock_guard lock(index_request->mutex_); }
+  indexer_waiter->cv.notify_all();
```

Comment:
It’s better to protect the notify with the mutex. If the thread is woken up for any reason, the signal would be lost

Reply:
The code is correct. It is a bit obscure, however.